opennsa/setup.py

```diff
@@ -168,8 +177,11 @@ def setupServiceFactory(self):
         nsa_name = domain_name + ':nsa'
 
         # base url
-        base_protocol = 'https://' if vc[config.TLS] else 'http://'
-        base_url = base_protocol + vc[config.HOST] + ':' + str(vc[config.PORT])
+        if vc[config.BASE_URL]:
+            base_url = vc[config.BASE_URL]
+        else:
+            base_protocol = 'https://' if vc[config.TLS] else 'http://'
+            base_url = base_protocol + vc[config.HOST] + ':' + str(vc[config.PORT])
```

Comment:
instead of having two patterns, why not simply deprecate the 'port + host' config, and just replace it with a BaseURL config... Or even better just get rid of port and make it so host requires a protocol to precede it. 

It seems cleaner and you can determine the port based on the prefix if need be.  

Wouldn't that support both your use case and the 'normal' use case?

Reply:
Requiring base_url in the config file will break all existing deployed configs.

If you want a non-default http or https port you do need a way to specify that.

Reply:
Fair enough, It just seems like you'll get confusion long term having two potential patterns to configure essentially the same thing.

Reply:
If it is any consolation most other applications in the NSI stack like nsi-safnari, nsi-pce, nsi-dds and nsi-requester use the same somewhat confusing way of configuring this.